src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -165,6 +165,17 @@ class ArrayBuffer[A] private (initElems: Array[AnyRef], initSize: Int)
     if(l > 0) Array.copy(array, 0, xs, start, l)
     xs
   }
+
+  /** Sorts this $coll in place according to an Ordering.
+    *
+    * @see [[scala.collection.mutable.IndexedSeqOps.sortInPlace]]
+    * @param  ord the ordering to be used to compare elements.
+    * @return modified input $coll sorted according to the ordering `ord`.
+    */
+  override def sortInPlace[B >: A]()(implicit ord: Ordering[B]): this.type = {
+    if (length > 1) scala.util.Sorting.stableSort(array.asInstanceOf[Array[B]])
+    this
+  }
```

Comment:
It is not totally clear to me how this will behave with `null` entries (elements of the underlying `Array` that are not yet used). Can you check that this does not throw an exception?

Reply:
@julienrf 
```scala
java.util.Arrays.sort(Array(null, "a"), implicitly[Ordering[String]])
```
throws NPE. So does 
```scala
Array(null, "a").sorted
```
 so nothing new here?

Reply:
`ArrayBuffer(3, 2, 1).sortInPlace` should not throw, can you try that?

Reply:
`scala.util.Sorting.stableSort` should allow sorting on a range within an array.  I had to rewrite the code on an emergency basis due to a license problem, so I didn't think too hard about the API at the time, but it looks like I left a private `mergeSort` method that does take array indices.  So it should be too hard to add methods to `scala.util.Sorting` that call in with indices that are shorter than the array length.

Reply:
I agree with @Ichoran. This is slightly beyond the scope of the PR but that would be great to just add an overload of `stableSort` that takes an array and indices and that calls the corresponding `Arrays.sort` method of the JDK. And we would use that one for `ArrayBuffer`.